### ISSUE TITLE
Problems with building dynamically

### ISSUE THREAD

**Issue by maneandrea** [IHaskell/IHaskell#1363]
I would like to build this dynamically so that I can use the libraries that I have already installed on my system. This is also in preparation of an AUR package that I am writing.

If I follow these steps:

```
$ git clone https://github.com/IHaskell/IHaskell.git
$ cd IHaskell
$ runhaskell Setup configure --ghc-option='-dynamic' --prefix=/usr --enable-tests
$ runhaskell Setup build
```
It builds fine (and much muuch faster than using stack!). However when I run the test suite:

```
$ runhaskell Setup test
```

I get the following output [pastebin.com/vc3rZ6UK](https://pastebin.com/vc3rZ6UK). All failures have this message in common:
```
       uncaught exception: SourceError
       Not in scope: ‘IHaskell.Display.displayFromChanEncoded’
       No module named ‘IHaskell.Display’ is imported.
```
and tests start failing from `Code Evaluation` onwards.

In the same pastebin I also indicate the output of
```
$ runhaskell Setup show-build-info
```

And also all the haskell packages that are installed in my system. Note that some of those packages do not exist in the archlinux repositories because I have built them myself (specifically ghc-parser, ipython-kernel, zeromq4 and semigroups). I am using the most up-to-date archlinux distribution with `ghc 9.0.2`.

**Comment by vaibhavsagar** [comment]
You might need these two lines before running the tests: https://github.com/IHaskell/IHaskell/blob/e0c1a8d3fd0589ea41e23d58d8ebc5a05480eb81/release.nix#L27-L28

This is because the IHaskell tests import and use the `ihaskell` library within an interpreter session. Let me know if that fixes it, or at least leads to a more interesting error. We recently merged https://github.com/IHaskell/IHaskell/pull/1360 so it would be wonderful if this works.

**Comment by maneandrea** [comment]
The same tests fail, but now with the message:
```
       uncaught exception: SourceError
       Bad interface file: /build/haskell-ihaskell/src/ihaskell-0.10.2.2/dist/build/IHaskell/Display.hi
           mismatched interface file ways (wanted "", got "dyn")
```

By the way: if I install ihaskell built in this way (ignoring the fact that the tests have failed) and I open it inside jupyterlab, I do get the  same error messages in the console.

**Comment by maneandrea** [comment]
One thing that sounds suspicious to me is that if I don't include the package [`ghc-static`](https://archlinux.org/packages/community/x86_64/ghc-static/) then the tests fail even worse:

```
       uncaught exception: SourceError
       Could not find module ‘Prelude’
       There are files missing in the ‘base-4.15.1.0’ package,
       try running 'ghc-pkg check'.
       Use -v (or `:set -v` in ghci) to see a list of the files searched for.
       ...
       Bad interface file: /build/haskell-ihaskell/src/ihaskell-0.10.2.2/dist/build/IHaskell/Display.hi
           mismatched interface file ways (wanted "", got "dyn")
       Bad interface file: /build/haskell-ihaskell/src/ihaskell-0.10.2.2/dist/build/IHaskell/IPython/Stdin.hi
           mismatched interface file ways (wanted "", got "dyn")
       Bad interface file: /build/haskell-ihaskell/src/ihaskell-0.10.2.2/dist/build/IHaskell/Eval/Widgets.hi
           mismatched interface file ways (wanted "", got "dyn")

```
So, somehow, `ihaskell` is looking for statically built libraries even if I pass the `-dynamic` flag.

**Comment by vaibhavsagar** [comment]
Hmm, I'm out of my depth here, sorry I can't be of more help.

**Comment by maneandrea** [comment]
A happy update. I restarted working on this after a while and the problem seems to have solved itself. In the meantime several archlinux haskell packages were updated or created, so that might have been the solution.

In any case, here is the AUR package, in case you have some archlinux friends that want to use it: [aur.archlinux.org/packages/ihaskell-git](https://aur.archlinux.org/packages/ihaskell-git)
